src/main/java/net/flintmc/gradle/manifest/tasks/GenerateFlintManifestTask.java

```diff
@@ -379,6 +426,53 @@ public void generate() {
           artifact.getClassifier() == null ? "" : "-" + artifact.getClassifier()
       );
 
+      SimpleMavenRepository internalRepository = this.flintGradlePlugin.getInternalRepository();
+      MavenArtifactDownloader downloader = this.flintGradlePlugin.getDownloader();
+
+      RemoteMavenRepository remoteMavenRepository;
+      if (flintGradlePlugin.getHttpClient() == null) {
+        remoteMavenRepository = null;
+      } else {
+        remoteMavenRepository = new RemoteMavenRepository(flintGradlePlugin.getHttpClient(), entry.getValue());
+      }
+
+      if (!internalRepository.isInstalled(artifact)) {
+        // The artifact is not installed already, install it
+        if (remoteMavenRepository == null) {
+          // Can't download anything in offline mode
+          throw new RuntimeException("Missing artifact " + artifact + " in local repository, " +
+              "but working in offline mode");
+        }
+
+        boolean setupSource = !downloader.hasSource(remoteMavenRepository);
+        if (setupSource) {
+          // The download has the source not set already, add it now
+          downloader.addSource(remoteMavenRepository);
+        }
+
+        try {
+          // Install the artifact including dependencies
+          downloader.installArtifact(artifact, internalRepository);
+        } catch (IOException e) {
+          throw new RuntimeException("Failed to install execution artifact", e);
+        }
+
+        if (setupSource) {
+          // We added the source, clean up afterwards
+          downloader.removeSource(remoteMavenRepository);
+        }
+      }
+
+      if (!artifactChecksums.has(artifact)) {
+        try (InputStream inputStream = internalRepository.getArtifactStream(artifact)) {
+          MessageDigest messageDigest = MessageDigest.getInstance("SHA-1");
+          String sha1Hex = Util.sha1Hex(messageDigest.digest(IOUtils.toByteArray(inputStream)));
+          this.artifactChecksums.add(artifact, sha1Hex);
+        } catch (IOException | NoSuchAlgorithmException e) {
+          throw new RuntimeException(e);
+        }
```

Comment:
Factor this out into the utilities class and don't throw runtime exceptions here.